test/interpreter/function-vars.spec.ts

```diff
@@ -0,0 +1,26 @@
+import {ErrorType, HyperFormula} from '../../src'
+import {adr, detailedError} from '../testUtils'
+import {ErrorMessage} from '../../src/error-message'
+
+describe('Function VAR.S', () => {
+  it('should take at least two arguments', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=VAR.S()'],
+      ['=VAR.S(1)']
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('A2'))).toEqualError(detailedError(ErrorType.DIV_BY_ZERO))
+  })
+
+  it('should calculate variance (sample)', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=VAR.S(2, 3)'],
+      ['=VAR.S(2, 3, 4, TRUE(), FALSE(), "1",)'],
+      ['=VAR.S(B3:I3)', 2, 3, 4, true, false, 'a', '\'1', null],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqual(0.5)
+    expect(engine.getCellValue(adr('A2'))).toBeCloseTo(2.28571428571429, 6) //inconsistency with product #1
+    expect(engine.getCellValue(adr('A3'))).toEqual(1)
+  })
```

Comment:
Please add clear descriptions to the conditions we're checking

```
it('should ignore empty cells, logical values, text and error values if argument is a reference', () => {
      ['=VAR.S(B3:I3)', 2, 3, 4, true, false, 'a', '\'1', null, '#DIV/0'],
```

```
it('should count logical values and text representation of numbers that are arguments', () => {
      ['=VAR.S(2, 3, 4, TRUE(), FALSE(), "1",)'],
```

Reply:
Errors are not ignored.

Reply:
Inconsistency with the docs then? Not again...

Reply:
Where is it in the docs?

Reply:
"Empty cells, logical values, text, or **error values** in the array or reference are ignored." Source: https://support.microsoft.com/en-us/office/var-s-function-913633de-136b-449d-813e-65a00b2b990b

Reply:
Well, the implementation clearly propagates errors 🥇

Reply:
Ok, nothing to add here then 🤷 Thanks for checking